File: C++/Hackerrank/extraLongFact.py
# Question Link: https://www.hackerrank.com/challenges/extra-long-factorials/problem
def extraLongFactorials(num):
    if num == 1:
        return 1
    else:
        return num*extraLongFactorials(num-1)

if __name__ == "__main__":
    n = int(input())
    res = extraLongFactorials(n)

    print(res)

File: C++/Hackerrank/extraLongFact.cpp
// Question Link: https://www.hackerrank.com/challenges/extra-long-factorials/problem
#include <bits/stdc++.h>
using namespace std;

int extraLongFact(long long int num) {
    if(num == 1) {
        return 1;
    } else {
        return num * extraLongFact(num-1);
    }
}

int main() {
    long long int n;
    cin >> n;

    long long int res = 0;

    res = extraLongFact(n);
    cout << res;
}